src/main/java/seedu/trackermon/logic/parser/FindCommandParser.java

```diff
@@ -19,15 +29,59 @@
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
+        ArgumentMultimap argumentMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_STATUS, PREFIX_TAG);
+        boolean hasPrefix = false;
+        boolean hasNamePrefix = argumentMultimap.arePrefixesPresent(PREFIX_NAME);
+        boolean hasStatusPrefix = argumentMultimap.arePrefixesPresent(PREFIX_STATUS);
+        boolean hasTagPrefix = argumentMultimap.arePrefixesPresent(PREFIX_TAG);
+        String[] keywordsArr;
+
+        List<Predicate<Show>> predicateArrayList = new ArrayList<>();
+
+        // Based on prefix, add user input as keywords into predicateArrayList
+        // PredicateArrayList.add -> AND Operator
+        // Whole list of keywords -> OR Operator
+        if (hasNamePrefix) {
+            hasPrefix = true;
+            String input = argumentMultimap.getValue(PREFIX_NAME).get();
+            keywordsArr = getKeywords(input);
+            for (int i = 0; i < keywordsArr.length; i++) {
+                predicateArrayList.add(new NameContainsKeywordsPredicate(Arrays.asList(keywordsArr[i])));
+            }
+        }
+
+        if (hasStatusPrefix) {
+            hasPrefix = true;
+            String input = argumentMultimap.getValue(PREFIX_STATUS).get();
+            keywordsArr = getKeywords(input);
+            predicateArrayList.add(new StatusContainsKeywordsPredicate(Arrays.asList(keywordsArr)));
+        }
+
+        if (hasTagPrefix) {
+            hasPrefix = true;
+            List<String> input = argumentMultimap.getAllValues(PREFIX_TAG);
+            for (int i = 0; i < input.size(); i++) {
+                predicateArrayList.add(new TagsContainsKeywordsPredicate(Arrays.asList(input.get(i))));
+            }
+        }
+
+        // if no prefix, find acts as a general search based on 1 keyword,
+        // otherwise it acts as a narrow search based on prefix and keywords and implements an AND search
+        // https://stackoverflow.com/questions/24553761/how-to-apply-multiple-predicates-to-a-java-util-stream
```

Comment:
haha it is necessary to keep this link?

Reply:
Noted. I found it useful and left it there. Will remove it